src/plugins/kernel_install/arch.sh

```diff
@@ -93,8 +93,9 @@ function generate_arch_temporary_root_file_system()
     fi
   fi
 
-  # We do not support initramfs outside grub scope
-  [[ "$bootloader_type" != 'GRUB' ]] && return
+  # We do not support initramfs outside grub and systemd-boot scope
+  [[ "$bootloader_type" != 'GRUB' && "$bootloader_type" != 'SYSTEMD_BOOT' ]] && return
```

Comment:
Wait, what? Systemd does use initramfs? Does it use something else?

Reply:
The systemd-boot (and grub and the efi firmware) could boot a unified kernel image which embeds the initramfs and the kernel in the same efi executable, but it is not mandatory and would need more changes to the kw deploy system to generate those file.